modules/packages/BLAS.chpl

```diff
@@ -19,103 +19,130 @@
+   matrix-matrix operations. Because the BLAS are efficient, portable, and
+   widely available, they are commonly used in the development of high quality
+   linear algebra software, LAPACK for example.
+
+This module wraps the functionality of level 3 matrix-matrix BLAS routines,
+supporting the array element types, ``real(32)`` (single), ``real`` (double),
+``complex(32)`` (complex), and ``complex`` (complex double) under a single
+interface.
+
+
 Compiling with BLAS
 -------------------
 
-In order to compile with the BLAS libraries, you will need
-BLAS and CBLAS (C wrappers to BLAS) installed on your system. You
-will also need to know the location of the cblas.h header file,
-as well as the BLAS libraries you need to link in (and where they
-are, if they're not in a standard location).
+In order to compile a Chapel program that uses this module, the
+BLAS and CBLAS (C wrappers to BLAS) libraries must be installed on the system.
+The paths to both the ``cblas.h`` header file and BLAS library
+must be passed to the ``-I`` and ``-L`` compiler arguments. The library name,
+typically ``blas``, must be passed to the ``-l`` argument as well.
 
-We recommend using a version of BLAS optimized for your system
-to get the best performance.
-
-As an example, with OpenBLAS :
+The compilation command should look something like this:
 
 .. code-block:: sh
 
     chpl -I$PATH_TO_CBLAS_DIR \
          -L$PATH_TO_BLAS_LIBS -lblas source.chpl
 
-You may need additional arguments for different libraries. For instance, if you
-use ATLAS, you will likely need to also add a -latlas to the above
-command. We strongly recommend referring to the documentation for the
-version of BLAS you have.
+BLAS Implementations:
+  There is a wide range of
+  `BLAS implementations <https://en.wikipedia.org/wiki/Basic_Linear_Algebra_Subprograms#Implementations>`_
+  available.
+  This module was built and tested with `netlib's CBLAS
+  <http://www.netlib.org/blas/#_cblas>`_, but many other implementations are
+  compatible as well.
+  Using a version of BLAS optimized for the user's system will yield the best
+  performance.
+
+  There are a few known caveats with some popular implementations:
+
+  * **ATLAS**
+
+    * The compilation command above likely requires the additional flag: ``-latlas``
+
+  * **MKL**
+
+    * The BLAS module assumes that the CBLAS functions are defined in the
+      header named ``cblas.h``.
+      MKL defines these in ``mkl_cblas.h``. This can be worked around by
+      creating a symbolic link to the correct header file name:
+      ``ln -s mkl_cblas.h cblas.h``
+
+  * **OpenBLAS**
+
+    * The header files that are included with OpenBLAS differ from the reference
+      CBLAS prototypes for complex arguments by using ``float*`` and ``double*``
+      pointers, instead of ``void*`` pointers.  Using this will likely result in
+      warnings about incompatible pointer types. These may be ignored.
```

Comment:
Do we want to also suggest using the cblas.h NETLIB header here, as a possible workaround?